### PR TITLE
Process Models Data - Blog App Project

### PR DESCRIPTION
**Pull request by Harrynoqs**
In this part, I created the models to process the app data

Workflow

Created empty models
Built the associations between the models
Created man

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+class Post < ApplicationRecord
+  after_create :update_posts_counter
```
Comment (alexander16108) [inline app/models/post.rb]: optional:
you should make the `:update_post_counter` method private and you should use the rails callback `after-save` to call the method as mentioned above every time a new post is saved to the database 👍🏾

File: app/models/like.rb [hunk 2]
```diff
@@ -0,0 +1,9 @@
+class Like < ApplicationRecord
+  after_create :update_likes_counter
```
Comment (alexander16108) [inline app/models/like.rb]: optional:
you should make the `:update_likes_counter` method private and you should use the rails callback `after-save` to call the method as mentioned above every time a new post is saved to the database 👍🏾

File: app/models/comment.rb [hunk 3]
```diff
@@ -0,0 +1,9 @@
+class Like < ApplicationRecord
+  after_create :update_likes_counter
```
Comment (alexander16108) [inline app/models/comment.rb]: optional:
you should make the `:update_comments_counter` method private and you should use the rails callback `after-save` to call the method mentioned above every time a new post is saved to the database 👍🏾

File: app/models/comment.rb [hunk 4]
```diff
@@ -0,0 +1,9 @@
...
+  belongs_to :post
+  belongs_to :author, class_name: 'User'
+
+  def update_likes_counter
+    Post.find(post.id).increment!(:likes_counter)
+  end
```
Comment (alexander16108) [inline app/models/comment.rb]: your comment model should have a method that updates the comments counter for a post. 😄

**Review by alexander16108 — CHANGES_REQUESTED** [review]
Hi @Harrynoqs , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights

📌 No Linters error ✅
📌 Descriptive pull request✅
📌 Good commit messages ✅
📌 Professional README file ✅

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by pinkmoon25 — APPROVED** [review]
Hi @Harrynoqs , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
Linters :heavy_check_mark: 
Good commits :heavy_check_mark: 
Gitflow :heavy_check_mark: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**